### PR TITLE
Update webpack to v3

### LINKED ISSUE #771: Optimize browser IDE's bundle size for production
### Rationale

Current bundle size is embarrassingly big. We do not minify it, there are several copies of same libs included, etc.

We should upgrade to webpack 3 and employ [all available optimisation techniques, new and old](https://github.com/webpack/docs/wiki/optimization)

### Affected modules and packages

`xod-client-browser`

### PR DESCRIPTION
**Pull request by brusherru**
It closes #771.

It made bundle size > x2 smaller (7.8MB -> 3.3MB), but it builds longer (even without uglifying) :(

### CODE REVIEW

File: packages/xod-client-browser/webpack.config.js [hunk 1]
```diff
@@ -1,88 +1,11 @@
...
-          'file?name=assets/font-awesome/[name].[ext]?[hash:6]',
-        ],
-      },
-      {
-        test: /\.json$/,
-        loader: 'json-loader',
-      },
-    ],
-  },
+const a = merge.smart(getBaseConfig(__dirname), {
```
Comment (nkrkv) [inline packages/xod-client-browser/webpack.config.js]: a?

File: packages/xod-client-electron/src/upload/components/PopupUploadProject.jsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+PopupUploadProject.defaultProps = {
+  upload: {
+    status: STATUS.STARTED,
+    message: '',
+    percentage: 0,
+  },
+  isVisible: false,
+};
+
+export default PopupUploadProject;
```
Comment (nkrkv) [inline packages/xod-client-electron/src/upload/components/PopupUploadProject.jsx]: Looks like a problem of rebase. AFAIK this file has gone after the deployment panel implementation.

File: packages/xod-client/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
+/**
+ * Basic Webpack Config that could be extended with additional
+ * configurations for specific platform (browser / electron).
+ *
+ * Because of Webpack config should contain some absolute paths (like `dist`),
+ * it's a function that accepts one argument: absolute path to specific platform
+ * root (`__dirname` in theirs webpack.config.js).
```
Comment (nkrkv) [inline packages/xod-client/webpack.config.js]: :+1: for the comment and humanism.

**Comment by brusherru** [comment]
@nkrkv @evgenykochetkov check it out, please

File: packages/xod-client-electron/package.json [hunk 4]
```diff
@@ -24,6 +24,7 @@
     "electron-settings": "^3.0.14",
     "electron-updater": "^2.7.1",
     "fs-extra": "^4.0.2",
+    "node-noop": "^1.0.0",
```
Comment (nkrkv) [inline packages/xod-client-electron/package.json]: :rofl: JS is so JS

Comment (evgenykochetkov) [inline packages/xod-client-electron/package.json]: 👍🏻

**Pull request merged**